lib/rules/no-eq-neg-zero.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * @fileoverview The rule should warn against code that tries to use the === operator to compare against -0.
+ * @author no-eq-neg-zero
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow use the === operator to compare against -0.",
+            category: "Possible Errors",
+            recommended: false
+        },
+        fixable: null,
+        schema: []
+    },
+
+    create(context) {
+        return {
+            BinaryExpression(node) {
+
+                /**
+                 * Checks a given node is -0
+                 *
+                 * @param {ASTNode} n - A node to check.
+                 * @returns {boolean} `true` if the node is -0.
+                 */
+                function eqNegZero(n) {
```

Comment:
Can you move this function outside of the `BinaryExpression` function, so that it's not redefined each time? Also, as a nitpick, can you use  a more descriptive parameter name than `n`?

```js
function eqNegZero(node) {
    // ...
}

module.exports = {
    meta: {
        // ...
    },
    create(context) {
        return {
            BinaryExpression(node) {
                // ...
            }
        }
    }
}
```